COMETwebapp/ViewModels/Components/ReferenceData/CategoryHierarchyDiagramViewModel.cs

```diff
@@ -0,0 +1,135 @@
+//    Copyright (c) 2023 RHEA System S.A.
+//
+//    Authors: Sam Gerené, Alex Vorobiev, Alexander van Delft, Jaime Bernar, Nabil Abbar
+//
+//    This file is part of COMET WEB Community Edition
+//    The COMET WEB Community Edition is the RHEA Web Application implementation of ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The COMET WEB Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or (at your option) any later version.
+//
+//    The COMET WEB Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//    Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+namespace COMETwebapp.ViewModels.Components.ReferenceData
+{
+    using System.Linq;
+
+    using Blazor.Diagrams.Core;
+    using Blazor.Diagrams.Core.Models;
+    using Blazor.Diagrams.Core.Geometry;
+
+    using CDP4Common.SiteDirectoryData;
+
+    using ReactiveUI;
+    using COMETwebapp.Components.ReferenceData;
+
+    /// <summary>
+    ///     View model for the <see cref="CategoryHierarchyDiagram" /> component
+    /// </summary>
+    public class CategoryHierarchyDiagramViewModel : ReactiveObject, ICategoryHierarchyDiagramViewModel
+    {
+        /// <summary>
+        ///     Backing field for <see cref="SelectedCategory" />
+        /// </summary>
+        private Category selectedCategory;
+
+        /// <summary>
+        ///     The selected <see cref="Category"/>
+        /// </summary>
+        public Category SelectedCategory
+        {
+            get => this.selectedCategory;
+            set => this.RaiseAndSetIfChanged(ref this.selectedCategory, value);
+        }
+
+        /// <summary>
+        /// A collection of <see cref="Category" />
+        /// </summary>
+        public IEnumerable<Category> Rows { get; set; }
+
+        /// <summary>
+        /// A collection of <see cref="Category" />
+        /// </summary>
+        public IEnumerable<Category> SubCategories { get; set; }
+
+        /// <summary>
+        /// The categories hierarchy <see cref="Diagram" /> to display
+        /// </summary>
+        public Diagram Diagram { get; set; }
+
+        /// <summary>
+        /// The svg path of arrow
+        /// </summary>
+        const string svgArrowPath = "M -0.093 17.86 V -18.5 L 29.233 -0.32 L -0.093 17.86 z M 1.407 -15.806 v 30.9715 L 26.3865 -0.32 L 1.407 -15.806 z";
```

Comment:
You can maybe create a SVG file with that contains and use it as a css class?

Reply:
We can't do that, the library uses svg path to create a LinkMarker